src/modules/charge-versions/controllers/charge-version-workflow.js

```diff
@@ -66,14 +59,21 @@ const postChargeVersionWorkflow = async request => {
  */
 const patchChargeVersionWorkflow = async (request, h) => {
   const { chargeVersionWorkflowId } = request.params;
-  const { approverComments, status } = request.payload;
+  const { approverComments, status, createdBy } = request.payload;
   const { chargeVersion } = request.pre;
 
   const changes = {
     ...request.payload,
     ...chargeVersion && { chargeVersion }
   };
-
+  if (createdBy) {
+    try {
+      changes.createdBy = userMapper.pojoToModel(createdBy);
```

Comment:
Does this need a try-catch? If we use the calling user found in `request.defra.internalCallingUser`, that data is taken from the IDM, so we can be sure that the data provided is in the correct format

This is the approach that is taken when creating charge versions from workflows in `controllers/charge-versions.js`

Reply:
The reason why we can't use request.defra object is because this endpoint can be used to update the charge version workflow record at anytime of the process so it will overwrite the original creator. I use the defra.user object in the UI though to get this info for the first submission of the draft charge info